### ISSUE TITLE
Add client ctor accepting neo config struct

### ISSUE THREAD

**Issue by gavin-norman-sociomantic** [sociomantic-tsunami/dhtproto#14]
Construction will then look like:
```D
auto dht = new DhtClient(epoll, legacy_config, neo_config);
auto dht_legacy_only = new DhtClient(epoll, legacy_config);
```

Requires sociomantic-tsunami/swarm#96 (swarm v4.3.0).

**Comment by gavin-norman-sociomantic** [comment]
(The new ctors will need to be wrapped in a `static if ( hasFeaturesFrom!("swarm", 4,3) )` block.)

**Comment by gavin-norman-sociomantic** [comment]
>(The new ctors will need to be wrapped in a static if ( hasFeaturesFrom!("swarm", 4,3) ) block.)

Actually, no. The neo branch isn't following semver yet. Breaking changes can occur at any point.

**Comment by gavin-norman-sociomantic** [comment]
This has been implemented (in #40) as a pure new feature.

**Comment by gavin-norman-sociomantic** [comment]
Fixed by #40.
